conduit/plugins/importers/algod/algod_importer.go

```diff
@@ -80,7 +82,120 @@ func init() {
 	}))
 }
 
-func (algodImp *algodImporter) Init(ctx context.Context, cfg plugins.PluginConfig, logger *logrus.Logger) (*sdk.Genesis, error) {
+func parseCatchpointRound(catchpoint string) (round sdk.Round, err error) {
+	parts := strings.Split(catchpoint, "#")
+	if len(parts) != 2 {
+		err = fmt.Errorf("unable to parse catchpoint, invalid format: %s", catchpoint)
+		return
+	}
+	uiRound, err := strconv.ParseUint(parts[0], 10, 64)
+	if err != nil {
+		return
+	}
+	round = sdk.Round(uiRound)
+	return
+}
+
+func (algodImp *algodImporter) startCatchpointCatchup() error {
+	// Run catchpoint catchup
+	client, err := common.MakeClient(algodImp.cfg.NetAddr, "X-Algo-API-Token", algodImp.cfg.CatchupConfig.AdminToken)
+	if err != nil {
+		return fmt.Errorf("received unexpected error creating catchpoint client: %w", err)
+	}
+	var resp string
+	err = client.Post(
+		algodImp.ctx,
+		&resp,
+		fmt.Sprintf("/v2/catchup/%s", common.EscapeParams(algodImp.cfg.CatchupConfig.Catchpoint)...),
+		nil,
+		nil,
+		nil,
+	)
+	if err != nil {
+		return fmt.Errorf("POST /v2/catchup/%s received unexpected error: %w", algodImp.cfg.CatchupConfig.Catchpoint, err)
+	}
+	return nil
+}
+
+func (algodImp *algodImporter) monitorCatchpointCatchup() error {
+	// Delay is the time to wait for catchup service startup
+	var delay = 5 * time.Second
+	start := time.Now()
+	// Report progress periodically while waiting for catchup to complete
+	running := true
+	for running {
+		select {
+		case <-time.After(delay):
+		case <-algodImp.ctx.Done():
+			return algodImp.ctx.Err()
+		}
+		stat, err := algodImp.aclient.Status().Do(algodImp.ctx)
+		if err != nil {
+			return fmt.Errorf("received unexpected error getting node status: %w", err)
+		}
+		running = stat.Catchpoint != ""
+		switch {
+		case !running:
+			break
+		case stat.CatchpointAcquiredBlocks > 0:
+			algodImp.logger.Infof("catchup phase 3 of 4 (Acquired Blocks): %d / %d", stat.CatchpointAcquiredBlocks, stat.CatchpointTotalBlocks)
+		case stat.CatchpointVerifiedAccounts > 0:
+			algodImp.logger.Infof("catchup phase 2 of 4 (Verified Accounts): %d / %d", stat.CatchpointVerifiedAccounts, stat.CatchpointTotalAccounts)
+		case stat.CatchpointProcessedAccounts > 0:
+			algodImp.logger.Infof("catchup phase 1 of 4 (Processed Accounts): %d / %d", stat.CatchpointProcessedAccounts, stat.CatchpointTotalAccounts)
+		default:
+			// Todo: We should expose catchupService.VerifiedBlocks via the NodeStatusResponse
+			algodImp.logger.Infof("catchup phase 4 of 4 (Verified Blocks)")
+		}
```

Comment:
How does this logging look? I think it changed a little with the KvMods.

Reply:
Looks like this
```
{"__type":"Conduit","_name":"main","level":"info","msg":"Starting Pipeline Initialization","time":"2023-03-27T07:19:54-07:00"}
{"__type":"Conduit","_name":"main","level":"info","msg":"Overriding default next round from 25101261 to 25101001.","time":"2023-03-27T07:19:54-07:00"}
{"__type":"importer","_name":"algod","level":"info","msg":"catchup phase 3 of 4 (Acquired Blocks): 7 / 1321","time":"2023-03-27T07:19:59-07:00"}
{"__type":"importer","_name":"algod","level":"info","msg":"catchup phase 3 of 4 (Acquired Blocks): 33 / 1321","time":"2023-03-27T07:20:04-07:00"}
{"__type":"importer","_name":"algod","level":"info","msg":"catchup phase 3 of 4 (Acquired Blocks): 62 / 1321","time":"2023-03-27T07:20:09-07:00"}
```
...
```
{"__type":"importer","_name":"algod","level":"info","msg":"Catchpoint catchup finished in 1m55.0583125s","time":"2023-03-27T07:21:49-07:00"}
```